### ISSUE TITLE
DataManager gets lost when providing Sprite as arg in Timer

### ISSUE THREAD

**Issue by NicoEnking** [photonstorm/phaser#4378]
When I set up a sprite and it's DataManager via `sprite.setDataEnabled()` and then set up a timer which has the Sprite as an arg, the "data" prop is undefined inside the callback for the Sprite provided as an arg.

```
let sprite = this.physics.add.sprite(x, y, "foo");
sprite.setDataEnabled(); // "data" prop is set from here on
console.log(sprite.data); // DataManager

let cb = function (sprite) {
  console.log(sprite.data); // undefined
}

this.time.addEvent({
  delay: 1000,
  callback: cb,
  callbackScope: this,
  args: [sprite]
});
```

**Comment by NicoEnking** [comment]
Does work in minimal setup. I need to check out why it does not work in my code.

**Comment by FMSArtist16** [comment]
On Feb 17, 2019 12:40 PM, "Nico Enking" <notifications@github.com> wrote:

Closed #4378 <https://github.com/photonstorm/phaser/issues/4378>.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/photonstorm/phaser/issues/4378#event-2144801699>, or mute
the thread
